### ISSUE TITLE
yamllint config with 'level': 'warning', will trigger fatal violation

### ISSUE THREAD

**Issue by konstruktoid** [ansible/ansible-lint#1794]
##### Summary

Even if the custom `.yamllint.yml` is read, the `'line-length': {'level': 'warning', 'max': 120, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}` rule will trigger an fatal violation.

`yamllint` itself triggers only a warning.

See https://github.com/konstruktoid/ansible-role-hardening/runs/4732728069?check_suite_focus=true

`.yamllint.yml`:
```yaml
---
extends: default
ignore: ".tox*\n.git*"
rules:
  line-length:
    max: 120
    level: warning
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```
$ ansible-lint --version
[WARNING]: You are running the development version of Ansible. You should only
run Ansible from "devel" if you are modifying the Ansible engine, or trying out
features under development. This is a rapidly changing source of code and can
become unstable at any point.
ansible-lint 5.3.2.dev5 using ansible 2.13.0.dev0
$ ansible-lint -vv
[...]
WARNING  Loading custom .yamllint.yml config file, this extends our internal yamllint config.
DEBUG    Effective yamllint rules used: {'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 2}, 'comments-indentation': {'level': 'warning'}, 'document-end': False, 'document-start': {'level': 'warning', 'present': True}, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'warning', 'max': 120, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': False, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false'], 'check-keys': True}}
[...]
WARNING  Listing 4 violation(s) that are fatal
yaml: line too long (143 > 120 characters) (line-length)
defaults/main/sshd.yml:20

yaml: line too long (122 > 120 characters) (line-length)
tasks/auditd.yml:21

yaml: line too long (129 > 120 characters) (line-length)
tasks/packagemgmt.yml:164

yaml: line too long (162 > 120 characters) (line-length)
tasks/packagemgmt.yml:192

You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - yaml  # Violations reported by yamllint

Finished with 4 failure(s), 0 warning(s) on 121 files.
```

```sh
$ yamllint -v
yamllint 1.26.3
$ # -s, --strict          return non-zero exit code on warnings as well as errors
$ if yamllint -s tasks/ defaults/; then echo "all ok"; fi
tasks/packagemgmt.yml
  164:121   warning  line too long (129 > 120 characters)  (line-length)
  192:121   warning  line too long (162 > 120 characters)  (line-length)

tasks/auditd.yml
  21:121    warning  line too long (122 > 120 characters)  (line-length)

defaults/main/sshd.yml
  20:121    warning  line too long (143 > 120 characters)  (line-length)

$ if yamllint tasks/ defaults/; then echo "all ok"; fi
tasks/packagemgmt.yml
  164:121   warning  line too long (129 > 120 characters)  (line-length)
  192:121   warning  line too long (162 > 120 characters)  (line-length)

tasks/auditd.yml
  21:121    warning  line too long (122 > 120 characters)  (line-length)

defaults/main/sshd.yml
  20:121    warning  line too long (143 > 120 characters)  (line-length)

all ok
```
